app/models/program.rb

```diff
@@ -94,4 +95,29 @@ def create_difficulty_levels
                                                 color: '#EF6E69')
     true
   end
+
+  ##
+  # Check if languages string has the right format. Used as before_save.
+  #
+  def check_languages_format
+    # All white spaces are removed to allow languages to be separated by ',' and ', ' but only the first option is saved
+    return unless self.languages.present?
+    self.languages = self.languages.delete(' ')
+    # We check if every language is a valid ISO 639-1 language
+    errors.add(:languages, 'must be two letters separated by commas') && return unless
+    self.languages.match(/^$|(\A[a-z][a-z](,[a-z][a-z])*\z)/).present?
+    error = false
+    self.languages.split(',').each do |lan|
```

Comment:
I just looked briefly into it but you could rewrite it as

```
self.languages.split(',').each do |lan|
  if ISO_639.find(lan).nil?
     errors.add(:languages, 'must be ISO 639-1 valid codes')
     return
end
```
Then you don't need the variable and the extra check after the loop!